alica_engine/src/engine/BasicBehaviour.cpp

```diff
@@ -180,31 +116,15 @@ void BasicBehaviour::doTrigger()
     _engine->editScheduler().schedule(std::bind(&BasicBehaviour::runJob, this, nullptr));
 }
 
-void BasicBehaviour::terminateJob()
+void BasicBehaviour::doTerminate()
 {
-    // There will be no run job if the interval is 0 or init is not executed
-    if (_activeRunJobId != -1) {
-        _engine->editScheduler().cancelJob(_activeRunJobId);
-        _activeRunJobId = -1;
-    }
     // Just to be double safe in terms of the correct behaviour of isSuccess() & isFailure() ensure result is reset before incrementing _execState
     _behResult.store(BehResult::UNKNOWN);
-    ++_execState;
-
-    clearFlags(Flags::RUN_TRACED);
-    if (!areFlagsSet(Flags::INIT_EXECUTED)) {
-        // Reset the execution context so that the RunningPlan instance can be deleted
-        _execContext.store(nullptr);
-        return;
-    }
-    clearFlags(Flags::INIT_EXECUTED);
+    setTerminatedState();
```

Comment:
this still won't work because the cancelJob needs to be done before the behaviour result is set to unknown, otherwise the run() method may overwrite the behaviour result. FYI: CancelJob blocks until run() completes. The order of the code is rather strict because we are using atomics so best to keep it as is.
You could instead split into: cancelJob -> set behaviour result -> setTerminatedState